File: libs/shared/src/client/cache.cc
#include "hestia/client/cache.h"

#include "session.h"

namespace hestia::client {
    proto::CacheInfo::Result Cache::info() {
        return session_->call<proto::CacheInfo>();
    }

    std::vector<proto::CacheEntry> Cache::list() {
        return session_->call<proto::CacheList>().entries;
    }

    proto::CacheUsage Cache::clear() {
        return session_->call<proto::CacheClear>();
    }
} // namespace hestia::client

File: crates/client/src/facades/cache.rs
use ipc::errors::IpcError;

use crate::session::Session;

pub struct Cache<'a> {
    pub(crate) session: &'a Session,
}

impl Cache<'_> {
    pub async fn info(&self) -> Result<proto::cache::CacheInfoResult, IpcError> {
        self.session
            .call::<proto::cache::CacheInfo>(&proto::Empty {})
            .await
    }

    pub async fn list(&self) -> Result<Vec<proto::cache::CacheEntry>, IpcError> {
        Ok(self
            .session
            .call::<proto::cache::CacheList>(&proto::Empty {})
            .await?
            .entries)
    }

    pub async fn clear(&self) -> Result<proto::cache::CacheUsage, IpcError> {
        self.session
            .call::<proto::cache::CacheClear>(&proto::Empty {})
            .await
    }
}
